src/main/java/org/sonarsource/auth/github/GitHubIdentityProvider.java

```diff
@@ -124,6 +137,54 @@ private GsonUser getUser(OAuthService scribe, Token accessToken) {
     return GsonTeams.parse(responseBody);
   }
 
+  public boolean isOrganizationMembershipRequired() { return settings.organizations().length > 0; }
+
+  private boolean isOrganizationsMember(Token accessToken, String login) {
+    for (String organization : settings.organizations()) {
+      if (isOrganizationMember(accessToken, organization, login)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private boolean isUnauthorized(Token accessToken, String login) {
+    return isOrganizationMembershipRequired() && !isOrganizationsMember(accessToken, login);
+  }
+
+  /**
+   * Check to see that login is a member of organization.
+   *
+   * A 204 response code indicates organization membership.  302 and 404 codes are not treated as exceptional,
```

Comment:
Honestly, I'm not 💯 sure.  https://developer.github.com/v3/orgs/members/#response-if-requester-is-an-organization-member-and-user-is-a-member makes it sound like a `404` can be returned in both of these cases:

> Response if requester is an organization member and user is not a member
> Response if requester is not an organization member and is inquiring about themselves

Neither of which mention the organization itself not existing... it's a strange API.